### PR TITLE
Issue51 no need for dataset parameter model at all

### LINKED ISSUE #51: Intermediate models DatasetParameter and VisualizationParameter seems to be obsolete
Intermediate models are needed only if there is extra information associated to the connection between two models. DatasetParameter and VisualizationParameter do not have such information and are therefore most likely obsolete.

### PR DESCRIPTION
**Pull request by aanersc**
closes #51 
@aperrin66 
@akorosov 

Visualization model is removed based on the new viewer. The only thing that remains from this issue#51 is the **DatasetParameter** model which should be checked. as mentioned in #51, there is no need to have the DatasetParameter in the geospaas. All the related content to it is removed and a new migration file is added. The procedure that has been done in this test can be done by django filtering with comma if needed:
https://github.com/nansencenter/django-geo-spaas/blob/704715d53537d4fcda5111d01a9ba278e67e003c/geospaas/catalog/tests.py#L125
We can search through manytomany relationship in django in both direction and that would cause no problem for us to lack DatasetParameter model for this kinds of searching.

Intermediatemodels are only needed when we want to have **extra information** for each of those manytomany relationship which is not the case for us and for django-geospaas parameters and datasets. It is any a simple manytomany relation here for these two:
image
Here we have only these two. It means it is pointless to have this model!

There is also one line of code in the harvesting repo that should also be removed. That line is also completely useless and no more needed. I will create a separate and issue for that.

### CODE REVIEW

File: geospaas/catalog/migrations/0001_initial.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='dataset',
             name='parameters',
-            field=models.ManyToManyField(through='catalog.DatasetParameter', to='vocabularies.Parameter'),
+            field=models.ManyToManyField(to='vocabularies.Parameter'),
```
Comment (aperrin66) [inline geospaas/catalog/migrations/0001_initial.py:106]: Did you modify that manually?

**Review by akorosov — CHANGES_REQUESTED** [review]
You cannot modify old migrations manually. You should only create new migrations.
See a similar issue here and apply this solution:
https://stackoverflow.com/a/39223849

**Comment by aanersc** [comment]
@aperrin66 
@akorosov 
No hand-made modification in the migration file! This is done by using the django 'makemigrations' and slightly modifying the autogenerated migration file no.9 in order to successfully migrate as shown in the link of @akorosov  in the above message!

It is ready to merge if you agreed!   :)

**Comment by aanersc** [comment]
@aperrin66 
@akorosov 
I have tested the tests of both harvesting repo and geospaas repo. It is working. And also I run the harvest.py and it is harvesting without any problem after migrate to the new one. In this migration file, first I remove the intermediary model then create m2m relationship in c519004.

**Review by aperrin66 — APPROVED** [review]
It's ok for me, but existing relationships between parameters and datasets will be lost.
This might be a problem for some users, so:
- the next release needs to be a major version.
- it must be made very clear, in the release notes, what will happen when applying this migration.

**Review by akorosov — APPROVED** [review]

**Pull request merged**